obspy/io/gse2/core.py

```diff
@@ -107,27 +107,25 @@ def _write_gse2(stream, filename, inplace=False, **kwargs):  # @UnusedVariable
             libgse2.write(trace.stats, trace.data, f, inplace)
 
 
-def _is_gse1(filename):
+def _is_gse1(file):
     """
     Checks whether a file is GSE1 or not.
 
-    :type filename: str
-    :param filename: GSE1 file to be checked.
+    :type file: str or file-like object
+    :param file: GSE1 file to be checked.
     :rtype: bool
     :return: ``True`` if a GSE1 file.
     """
-    # Open file.
-    with open(filename, 'rb') as f:
+    with open_bytes_stream(file) as stream:
         try:
-            data = f.readline()
+            data = stream.readline()
         except Exception:
             return False
-    if data.startswith(b'WID1') or data.startswith(b'XW01'):
-        return True
-    return False
+        return data.startswith(b'WID1') \
+            or data.startswith(b'XW01')
```

Comment:
This return statement can be dedented, no reason to have it inside the `with` block